src/cpp/src/model_runner.hpp

```diff
@@ -153,6 +171,7 @@ class ModelRunner {
                 subsequence_begins_data += 1;
                 block_indices_begins_data += 1;
             }
+            sequence_group->set_seq_len_to_sample(tokens_num_to_sample);
```

Comment:
looks like for case of multiple sequences within a group, you will `tokens_num_to_sample` multiplied by `num_running_sequences` (because for loop increments `tokens_num_to_sample` `num_running_sequences` times)